store/auth.go

```diff
@@ -42,6 +42,9 @@ type Authorizer interface {
 	// errors, as the higher-level code might as well treat Authorize
 	// as best-effort and only log any returned error.
 	Authorize(r *http.Request, dauthCtx DeviceAndAuthContext, user *auth.UserState, opts *AuthorizeOptions) error
+
+	// HasAuth returns whether is possible to authorize requests for user.
```

Comment:
```suggestion
	// HasAuth returns whether it is possible to authorize requests for user.
```
The comment is not very clear, and the various implementations are not very useful in understanding what this does. Does this check whether the user has valid credentials, or is it (as I guess looking at the code, but I might be wrong) a  partial check to see if in principle calling `Authorize` with this user as parameter can possibly work (that is, if `HasUser` returns `false` then a call to `Authoriza` will fail for sure, but even if `HasUser` returns `true`, the call to `Authorize` might still fail because the user — although having valid credentials — is not authorized to perform the specific request `r`)?

Reply:
I renamed this and reworked the comment, also adding comments its new implementations